### PR TITLE
[Build] Rust dependency for tests

### PR DESCRIPTION
**Pull request by riedgar-ms**
We need the Rust toolchain installed now. Take the opportunity to expand the context window on some tests, with the goal of making them fractionally more reliable.

### CODE REVIEW

File: .github/workflows/ci_tests.yml [hunk 1]
```diff
@@ -23,6 +23,12 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
```
[the remark this answers is not in the document]
Comment (riedgar-ms) [reply .github/workflows/ci_tests.yml:27]: I'm seeing alternatives such as:
https://github.com/marketplace/actions/setup-rust-toolchain-for-github-ci
https://github.com/marketplace/actions/set-up-a-rust-toolchain
Would one of these be better @paulbkoch

**Comment by riedgar-ms** [comment]
Took a few goes, but this finally went through CI:
https://github.com/guidance-ai/guidance/actions/runs/8765018844

File: .github/workflows/ci_tests.yml [hunk 2]
```diff
@@ -23,6 +23,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
+      - name: Install Rust
+        shell: bash
+        run: |
+           curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain 1.75.0
```
Comment (riedgar-ms) [inline .github/workflows/ci_tests.yml:29]: This was Michal's suggestion

**Comment by paulbkoch** [comment]
Thanks @riedgar-ms!  Sorry this was failing intermittently.  I guess we got lucky on the PR build then.

**Pull request merged**